jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/internal/ByteChannelContentSource.java

```diff
@@ -246,28 +246,24 @@ public boolean rewind()
 
     /**
      * A {@link ByteChannelContentSource} for a {@link Path}
-     * @deprecated To be replaced by an updated {@link org.eclipse.jetty.io.content.PathContentSource} in 12.1.0
      */
-    @Deprecated(forRemoval = true, since = "12.0.11")
     public static class PathContentSource extends ByteChannelContentSource
     {
         private final Path _path;
 
         public PathContentSource(Path path)
         {
-            super(new ByteBufferPool.Sized(null), null, 0, size(path));
-            _path = path;
+            this(ByteBufferPool.SIZED_NON_POOLING, path, 0, -1);
         }
 
         public PathContentSource(ByteBufferPool.Sized byteBufferPool, Path path)
         {
-            super(byteBufferPool, null, 0, size(path));
-            _path = path;
+            this(byteBufferPool, path, 0, -1);
         }
 
         public PathContentSource(ByteBufferPool.Sized byteBufferPool, Path path, long offset, long length)
         {
-            super(byteBufferPool, null, offset, length);
+            super(byteBufferPool, null, offset, length < 0 ? size(path) : length);
```

Comment:
Yeah, I keep tossing over this one... should we just return length form the path each time getLength is called, but then it might not match the actual length read.    So, since we are returning a specific length, then I think it is right to fix the size and not seek to EOF.  If the file changes, then that will either be truncated or a read error.